### PR TITLE
Skeleton

### PR DESCRIPTION
**Pull request by juanimedone**
# Skeleton

## Overview

This pull request introduces the initial skeleton for the project, started from scratch. It is still not operational.

## Features

- **Register Management:** Functions for setting and getting register values.
- **Memory Access:** Methods for reading and writing to memory, including handling memory-mapped I/O.
- **Instruction Handling:** Framework for decoding and executing instructions.
- **Trap Operations:** Basic setup for handling trap routines.

## Future Work

- Fix bug/s to make the VM run properly.
- Add error handling and edge case management for various operations.
- Write unit tests to ensure correctness and stability.
- Add execution instructions in `README.md` and code documentation.

### CODE REVIEW

File: src/instructions.rs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    TRAP,   // execute trap
+}
+
+fn sign_extend(x: u16, bit_count: usize) -> u16 {
+    if (x >> (bit_count - 1)) & 1 != 0 {
+        x | (0xFFFF << bit_count)
+    } else {
+        x
+    }
+}
```
Comment (lazcanoluca) [inline src/instructions.rs]: Maybe this belongs in the `utils` module?

**Review by lazcanoluca — APPROVED** [review]
Looks very promising! I left a comment, just a nitpick.

File: src/instructions.rs (hunk 1 above)
Comment (juanimedone) [reply src/instructions.rs]: Yes, you're right.
